File: src/checks/balance.ts
import { Check, CheckContext, CheckResult } from '../types.js';
import { createClient, DripError } from '../drip-client.js';

export const balanceGetCheck: Check = {
  name: 'Balance Get',
  description: 'Get customer balance',
  async run(ctx: CheckContext): Promise<CheckResult> {
    const start = performance.now();
    const client = createClient(ctx);
    const customerId = ctx.createdCustomerId || ctx.testCustomerId;

    if (!customerId) {
      return {
        name: 'Balance Get',
        success: false,
        duration: 0,
        message: 'No customer ID available',
        suggestion: 'Run Customer Create check first or set TEST_CUSTOMER_ID',
      };
    }

    try {
      const result = await client.getBalance(customerId);
      const duration = performance.now() - start;

      return {
        name: 'Balance Get',
        success: true,
        duration,
        message: `$${result.balanceUsdc} USDC`,
      };
    } catch (error) {
      const duration = performance.now() - start;
      const err = error as DripError;
      return {
        name: 'Balance Get',
        success: false,
        duration,
        message: err.message || 'Failed to get balance',
        details: `Customer ID: ${customerId}`,
      };
    }
  },
};

File: python/checks/balance.py
"""Balance check."""
from ..types import Check, CheckContext, CheckResult
from ..drip_client import create_client


async def _balance_get_check(ctx: CheckContext) -> CheckResult:
    """Get customer balance."""
    customer_id = ctx.created_customer_id or ctx.test_customer_id
    if not customer_id:
        return CheckResult(
            name="balance_get",
            success=False,
            duration=0,
            message="No customer ID available",
            suggestion="Run customer_create check first"
        )

    try:
        client = create_client(ctx.api_key, ctx.api_url)
        balance = client.get_balance(customer_id)

        # Handle different response formats
        balance_usdc = getattr(balance, 'balance_usdc', getattr(balance, 'balance', 'N/A'))
        available_usdc = getattr(balance, 'available_usdc', getattr(balance, 'available', balance_usdc))

        return CheckResult(
            name="balance_get",
            success=True,
            duration=0,
            message=f"Balance: {balance_usdc} USDC",
            details=f"available: {available_usdc} USDC"
        )
    except Exception as e:
        return CheckResult(
            name="balance_get",
            success=False,
            duration=0,
            message=f"Failed to get balance: {e}"
        )


balance_get_check = Check(
    name="balance_get",
    description="Get customer balance",
    run=_balance_get_check
)
